aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/operation/AWSS3StorageGetUrlOperation.java

```diff
@@ -0,0 +1,97 @@
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package com.amplifyframework.storage.s3.operation;
+
+import android.annotation.SuppressLint;
+import androidx.annotation.NonNull;
+
+import com.amplifyframework.core.Consumer;
+import com.amplifyframework.storage.StorageException;
+import com.amplifyframework.storage.operation.StorageGetUrlOperation;
+import com.amplifyframework.storage.result.StorageGetUrlResult;
+import com.amplifyframework.storage.s3.request.AWSS3StorageGetUrlRequest;
+import com.amplifyframework.storage.s3.service.AWSS3StorageService;
+import com.amplifyframework.storage.s3.utils.S3RequestUtils;
+
+import com.amazonaws.mobile.client.AWSMobileClient;
+
+import java.net.URL;
+
+/**
+ * An operation to retrieve pre-signed object URL from AWS S3.
+ */
+public final class AWSS3StorageGetUrlOperation
+        extends StorageGetUrlOperation<AWSS3StorageGetUrlRequest> {
+    private final AWSS3StorageService storageService;
+    private final Consumer<StorageGetUrlResult> onResult;
+    private final Consumer<StorageException> onError;
+    private URL url;
+
+    /**
+     * Constructs a new AWSS3StorageGetUrlOperation.
+     * @param storageService S3 client wrapper
+     * @param request getUrl request parameters
+     * @param onSuccess Notified when URL is generated.
+     * @param onError Notified upon URL generation error
+     */
+    public AWSS3StorageGetUrlOperation(
+            @NonNull AWSS3StorageService storageService,
+            @NonNull AWSS3StorageGetUrlRequest request,
+            @NonNull Consumer<StorageGetUrlResult> onSuccess,
+            @NonNull Consumer<StorageException> onError) {
+        super(request);
+        this.storageService = storageService;
+        this.onResult = onSuccess;
+        this.onError = onError;
+        this.url = null;
+    }
+
+    @SuppressLint("SyntheticAccessor")
+    @Override
+    public void start() {
+        final String identityId;
+        final String serviceKey;
+
+        try {
+            // Obtain S3 service key for storage service
+            identityId = AWSMobileClient.getInstance().getIdentityId();
+            serviceKey = S3RequestUtils.getServiceKey(
+                    getRequest().getAccessLevel(),
+                    identityId,
+                    getRequest().getKey(),
+                    getRequest().getTargetIdentityId()
+            );
+
+            try {
+                this.url = storageService.getPresignedUrl(serviceKey, getRequest().getExpires());
+                onResult.accept(StorageGetUrlResult.fromUrl(url));
+            } catch (Exception exception) {
```

Comment:
I did this because I honestly couldn't figure out how `AmazonS3Client#generatePresignedUrl()` could ever fail. As long as the credentials are correctly setup in the first try-catch, I don't see a case where PresignedUrl generation would fail. I only try-caught it because it says it can throw `AmazonClientException`.